it/google-cloud-platform/src/main/java/com/google/cloud/teleport/it/gcp/datagenerator/DataGenerator.java

```diff
@@ -101,11 +101,16 @@ public Integer execute(Duration timeout) throws IOException {
       Result dataGeneratorResult = pipelineOperator.waitUntilDoneAndFinish(config);
       assertThatResult(dataGeneratorResult).hasTimedOut();
     }
-    int generatedMessages =
-        pipelineLauncher
-            .getMetric(
-                PROJECT, REGION, dataGeneratorLaunchInfo.jobId(), MESSAGES_GENERATED_METRIC_NAME)
-            .intValue();
+    int generatedMessages = 0;
+    try {
```

Comment:
could you please add a justification for this update in the PR description?
something like this: the method can throw an NPE exception. Added try-catch statement